controllers/propagator/policy_controller.go

```diff
@@ -124,6 +125,19 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 				return reconcile.Result{}, err
 			}
 
+			// Clean up hub template watch metric
+			log.V(3).Info("Deleting hub template watch metric")
+
+			metricDeleted := hubTemplateActiveWatchesMetric.Delete(prometheus.Labels{
```

Comment:
Do you think this would be better in `cleanUpPolicy`?

Reply:
I put it here so that it would be more visible alongside the other metric setting. I figured that way if we wanted to update it in the future, it'd all be in a single file and a single function.

Reply:
...but using `defer` covers the cleanup case also, right?

Reply:
Yup since we would just want it to be `0` when there are no more watches and not delete the metric. If we had a watch count per policy, then we would delete the metric when the policy gets deleted.